add MCP apps docs (#410)
kagent-dev/website: src/app/docs/kagent/concepts/mcp-apps/page.mdx

---
title: "MCP Apps"
pageOrder: 7
description: "Render interactive MCP UI widgets (MCP Apps) inline in the kagent chat and test them from the dashboard."
---

export const metadata = {
  title: "MCP Apps in kagent",
  description: "Render interactive MCP UI widgets (MCP Apps) inline in the kagent chat and test them from the dashboard.",
  author: "kagent.dev"
};

# MCP Apps

MCP Apps are MCP tools that ship an interactive user interface with their results. Instead of returning plain JSON, an MCP server can attach an HTML/UI **resource** to a tool using the MCP Apps extension ([`@modelcontextprotocol/ext-apps`](https://github.com/modelcontextprotocol/ext-apps)). When an agent calls such a tool, kagent renders a live widget — a form, board, chart, or dashboard — directly in the chat.

## How it works

> An MCP server advertises an app by setting `_meta.ui.resourceUri` on a tool (for example, `ui://server-everything/weather-dashboard`). kagent detects this metadata automatically — no extra configuration is needed:

Reviewer: > An MCP server advertises an app by setting `_meta.ui.resourceUri` on a tool (for example, `ui://server-everything/weather-dashboard`).

So I was expecting one of the config examples later to have this field, but I do not see it in the config.

Is this a backend field in the MCP server that users do not need to configure or see? if so, maybe tweak the description:
```suggestion
An MCP server advertises an app by setting `_meta.ui.resourceUri` on a tool in its configuration (`ui://server-everything/weather-dashboard`). kagent detects this metadata on the MCP server automatically: no extra kagent configuration is needed.
```
